consensus/parlia/fusionfork.go

```diff
@@ -0,0 +1,226 @@
+package parlia
+
+import (
+	"container/heap"
+	"context"
+	"math"
+	"math/big"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/systemcontracts"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// init contract
+func (p *Parlia) initFusionContract(state *state.StateDB, header *types.Header, chain core.ChainContext,
+	txs *[]*types.Transaction, receipts *[]*types.Receipt, receivedTxs *[]*types.Transaction, usedGas *uint64, mining bool,
+) error {
+	// method
+	method := "initialize"
+	// contracts
+	contracts := []string{
+		systemcontracts.StakeHubContract,
+	}
+	// get packed data
+	data, err := p.stakeHubABI.Pack(method)
+	if err != nil {
+		log.Error("Unable to pack tx for init fusion contract", "error", err)
+		return err
+	}
+	for _, c := range contracts {
+		msg := p.getSystemMessage(header.Coinbase, common.HexToAddress(c), data, common.Big0)
+		// apply message
+		log.Trace("init fusion contract", "block hash", header.Hash(), "contract", c)
+		err = p.applyTransaction(msg, state, header, chain, txs, receipts, receivedTxs, usedGas, mining)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type ValidatorItem struct {
+	address     common.Address
+	votingPower *big.Int
+}
+
+// An ValidatorHeap is a min-heap of validator's votingPower.
+type ValidatorHeap []ValidatorItem
+
+func (h ValidatorHeap) Len() int { return len(h) }
+
+// We want topK validators with max voting power, so we need a min-heap
+func (h ValidatorHeap) Less(i, j int) bool {
+	if h[i].votingPower.Cmp(h[j].votingPower) == 0 {
```

Comment:
I am afraid that `h[i].votingPower` can be nil.

Reply:
As I see, it will never be nil. Because the contracts will always return a valid value.